gslib/utils/boto_util.py

```diff
@@ -266,9 +266,18 @@ def GetNewHttp(http_class=httplib2.Http, **kwargs):
     An initialized httplib2.Http instance.
   """
 
-  # Get proxy info, will get None if no proxies are set in boto or
-  # environment variables
-  proxy_info = SetProxyInfo()
+  ##Get Proxy configuration from boto file, defaults are None, 0 and False
+  boto_proxy_config = {
+      'proxy_host': config.get('Boto', 'proxy', None),
+      'proxy_type': config.get('Boto', 'proxy_type', 'http'),
+      'proxy_port': config.getint('Boto', 'proxy_port'),
+      'proxy_user': config.get('Boto', 'proxy_user', None),
+      'proxy_pass': config.get('Boto', 'proxy_pass', None),
+      'proxy_rdns': config.get('Boto', 'proxy_rdns')
+  }
```

Comment:
It wouldn't interfere with the environment variables because of the code below, that gets executed after the aforementioned code above.

```
 if not (proxy_info.proxy_host and proxy_info.proxy_port):
    # Fall back to using the environment variable. Use only http proxies.
    for proxy_env_var in ['http_proxy', 'https_proxy', 'HTTPS_PROXY']:
      if proxy_env_var in os.environ and os.environ[proxy_env_var]:
        proxy_info = ProxyInfoFromEnvironmentVar(proxy_env_var)
        # Assume proxy_rnds is True if a proxy environment variable exists.
        proxy_info.proxy_rdns = config.getbool('Boto', 'proxy_rdns', True)
        break
```